pyspecProcScripts/correlation_alignment.py

```diff
@@ -117,47 +117,55 @@ def correl_align(
     energy_diff = 1.0
     i = 0
     energy_vals = []
-    this_E = (abs(s_orig.C.sum(indirect_dim)) ** 2).data.sum().item() / N**2
+    this_E = (abs(s_jk.C.sum(indirect_dim)) ** 2).data.sum().item() / N**2
     energy_vals.append(this_E / sig_energy)
     last_E = None
-    for_nu_center = s_orig.C
+    # {{{ find center frequency to see where to center the mask
+    for_nu_center = s_jk.C
     for_nu_center.ft(list(signal_pathway))
     for x in range(len(signal_keys)):
         for_nu_center = for_nu_center[signal_keys[x], signal_values[x]]
     nu_center = for_nu_center.mean(indirect_dim).C.argmax(direct)
+    # }}}
     logging.debug(psd.strm("Center frequency", nu_center))
+    s_jk.ift(direct)
     for my_iter in range(100):
         i += 1
         logging.debug(psd.strm("*** *** ***"))
         logging.debug(psd.strm("CORRELATION ALIGNMENT ITERATION NO. ", i))
         logging.debug(psd.strm("*** *** ***"))
-        s_orig.ift(direct)
-        s_copy = s_orig.C
+        # s_mn will become the signal with the additional Delta ph# dimension.
+        s_mn = s_jk.C
         if align_phases:
-            ph0 = s_orig.C.sum(direct)
+            ph0 = s_jk.C.sum(direct)
             ph0 /= abs(ph0)
-            s_copy /= ph0
-        s_copy.ft(direct)
+            s_mn /= ph0
+        # {{{ Apply mask around center of signal in nu domain
+        s_mn.ft(direct)
         this_mask = np.exp(
-            -((s_copy.fromaxis(direct) - nu_center) ** 2) / (2 * sigma**2)
+            -((s_mn.fromaxis(direct) - nu_center) ** 2) / (2 * sigma**2)
         )
-        s_copy *= this_mask
-        s_copy.ift(direct)
-        s_copy2 = s_orig.C
-        for k, v in ph_len.items():
-            ph = np.ones(v)
-            s_copy *= psd.nddata(ph, "Delta" + k.capitalize())
-            s_copy.setaxis("Delta" + k.capitalize(), "#")
-        correl = s_copy * 0
-        for k, v in ph_len.items():
-            for ph_index in range(v):
-                s_copy["Delta%s" % k.capitalize(), ph_index] = s_copy[
-                    "Delta%s" % k.capitalize(), ph_index
-                ].run(lambda x, axis=None: np.roll(x, ph_index, axis=axis), k)
-        for j in range(1, N):
```

Comment:
before making the other modifications to the code, did you check that the examples still ran with the mod that we have made together in the office? (removing the `range(1,N)` loop here?